executor/transaction_processor.go

```diff
@@ -120,6 +122,31 @@ type ethTestProcessor struct {
 
 // Process transaction inside state into given LIVE StateDb
 func (p *ethTestProcessor) Process(state State[txcontext.TxContext], ctx *Context) error {
+	// This check needs to be done before ApplyMessage is called, otherwise different state-root hash is produced
```

Comment:
maybe reformulate this to ... if this line of argument is correct in your opinion:
```suggestion
	// This check needs to be done before ApplyMessage is called to identify invalid transactions. Invalid
	// transactions are expected to be filtered out before running them (via ApplyMessage). If they would
	// get processed, they would at least update the nonce of the transaction sender, thereby influencing
	// the resulting state hash. This would be detected as a failed test case.
```